src/lib/swapper/swappers/LifiSwapper/endpoints.ts

```diff
@@ -19,17 +19,21 @@ import { createDefaultStatusResponse } from 'lib/utils/evm'
 
 import { getTradeQuote } from './getTradeQuote/getTradeQuote'
 import { getLifiChainMap } from './utils/getLifiChainMap'
+import { getLifiToolsMap } from './utils/getLifiToolsMap'
 import { getUnsignedTx } from './utils/getUnsignedTx/getUnsignedTx'
+import type { LifiTool } from './utils/types'
 
 const tradeQuoteMetadata: Map<string, Route> = new Map()
 
-let lifiChainMapPromise: Promise<Result<Map<ChainId, ChainKey>, SwapErrorRight>> | undefined
+// cached metadata - would need persistent cache with expiry if moved server-side
+let lifiChainMapPromise: Promise<Map<ChainId, ChainKey>> | undefined
+let lifiToolsMapPromise: Promise<Map<string, LifiTool>> | undefined
```

Comment:
Note, this will work for the "base" list of tools i.e mainnet tools and some tools on other EVM chains but Li.Fi API sometimes appends stuff depending on the chain, see

<img width="962" alt="image" src="https://github.com/shapeshift/web/assets/17035424/1bee2f4a-0398-43f9-8e73-cf44f816d00e">
<img width="837" alt="image" src="https://github.com/shapeshift/web/assets/17035424/ce1b1c25-33dd-4d61-bf0f-829502d7235d">

Reply:
6752a7b33

<img width="669" alt="image" src="https://github.com/shapeshift/web/assets/17035424/1e767d92-61fd-4973-b9ca-e97ca5483114">